spanner/api/Spanner.Samples/ListInstanceConfigsAsync.cs

```diff
@@ -19,12 +19,13 @@
 using System;
 using System.Collections.Generic;
 using System.Linq;
+using System.Threading.Tasks;
 
-public class ListInstanceConfigsSample
+public class ListInstanceConfigsAsyncSample
 {
-    public IEnumerable<InstanceConfig> ListInstanceConfigs(string projectId)
+    public async Task<IEnumerable<InstanceConfig>> ListInstanceConfigsAsync(string projectId)
     {
-        var instanceAdminClient = InstanceAdminClient.Create();
+        var instanceAdminClient = await InstanceAdminClient.CreateAsync();
         var projectName = ProjectName.FromProject(projectId);
         var instanceConfigs = instanceAdminClient.ListInstanceConfigs(projectName);
```

Comment:
If we are making the method Async, I think this might have an async version. Can you please double check?

Reply:
Here is my suggestion. 
Can I add the System.Linq.Parallel package to the project in order to print only 5 configurations to the console?

```suggestion
                var instanceConfigs = instanceAdminClient.ListInstanceConfigsAsync(projectName);
        
        // We print the first 5 elements for demonstration purposes.
        // You can print all configs in the sequence by removing the call to Take(5).
        // The sequence will lazily fetch elements in pages as needed.
        await foreach(var instanceConfig in instanceConfigs.Take(5))
        {
            Console.WriteLine($"Available leader options for instance config {instanceConfig.InstanceConfigName.InstanceConfigId}:");
            foreach (var leader in instanceConfig.LeaderOptions)
            {
                Console.WriteLine(leader);
            }

            Console.WriteLine($"Available optional replica for instance config {instanceConfig.InstanceConfigName.InstanceConfigId}:");
            foreach (var optionalReplica in instanceConfig.OptionalReplicas)
            {
                Console.WriteLine($"Replica type - {optionalReplica.Type}, default leader location - {optionalReplica.DefaultLeaderLocation}, location - {optionalReplica.Location}");
            }
        }

        return instanceConfigs.ToEnumerable();
```